pybricks/media/pb_type_image.c

```diff
@@ -0,0 +1,353 @@
+
+static mp_obj_t pb_type_Image_draw_line(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(x1),
+        PB_ARG_REQUIRED(y1),
+        PB_ARG_REQUIRED(x2),
+        PB_ARG_REQUIRED(y2),
+        PB_ARG_DEFAULT_INT(width, 1),
+        PB_ARG_DEFAULT_NONE(color));
+
+    mp_int_t x1 = pb_obj_get_int(x1_in);
+    mp_int_t y1 = pb_obj_get_int(y1_in);
+    mp_int_t x2 = pb_obj_get_int(x2_in);
+    mp_int_t y2 = pb_obj_get_int(y2_in);
+    mp_int_t width = pb_obj_get_int(width_in);
+    int color = get_color(color_in);
+
+    if (width <= 1) {
+        pbio_image_draw_line(&self->image, x1, y1, x2, y2, color);
+    } else {
+        pbio_image_draw_thick_line(&self->image, x1, y1, x2, y2, width, color);
+    }
+
+    if (self->is_display) {
+        pbdrv_display_update();
+    }
+
+    return mp_const_none;
+}
+static MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Image_draw_line_obj, 1, pb_type_Image_draw_line);
+
+static mp_obj_t pb_type_Image_draw_box(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(x1),
+        PB_ARG_REQUIRED(y1),
+        PB_ARG_REQUIRED(x2),
+        PB_ARG_REQUIRED(y2),
+        PB_ARG_DEFAULT_INT(r, 0),
+        PB_ARG_DEFAULT_FALSE(fill),
+        PB_ARG_DEFAULT_NONE(color));
+
+    mp_int_t x1 = pb_obj_get_int(x1_in);
+    mp_int_t y1 = pb_obj_get_int(y1_in);
+    mp_int_t x2 = pb_obj_get_int(x2_in);
+    mp_int_t y2 = pb_obj_get_int(y2_in);
+    mp_int_t r = pb_obj_get_int(r_in);
+    bool fill = mp_obj_is_true(fill_in);
+    int color = get_color(color_in);
+
+    int width = x2 - x1 + 1;
+    int height = y2 - y1 + 1;
+    if (!fill) {
+        pbio_image_draw_rounded_rect(&self->image, x1, y1, width, height, r, color);
+    } else {
+        pbio_image_fill_rounded_rect(&self->image, x1, y1, width, height, r, color);
+    }
+
+    if (self->is_display) {
+        pbdrv_display_update();
+    }
+
+    return mp_const_none;
+}
+static MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Image_draw_box_obj, 1, pb_type_Image_draw_box);
+
+static mp_obj_t pb_type_Image_draw_circle(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(x),
+        PB_ARG_REQUIRED(y),
+        PB_ARG_REQUIRED(r),
+        PB_ARG_DEFAULT_FALSE(fill),
+        PB_ARG_DEFAULT_NONE(color));
+
+    mp_int_t x = pb_obj_get_int(x_in);
+    mp_int_t y = pb_obj_get_int(y_in);
+    mp_int_t r = pb_obj_get_int(r_in);
+    bool fill = mp_obj_is_true(fill_in);
+    int color = get_color(color_in);
+
+    if (!fill) {
+        pbio_image_draw_circle(&self->image, x, y, r, color);
+    } else {
+        pbio_image_fill_circle(&self->image, x, y, r, color);
+    }
```

Comment:
```suggestion
    if (fill) {
        pbio_image_fill_circle(&self->image, x, y, r, color);
    } else {
        pbio_image_draw_circle(&self->image, x, y, r, color);
    }
```

Minor nit: reads a bit easier without the `!` to my eyes.

Reply:
For me, the no-fill is the default normal case, that’s why I put it first, but OK, I can make the change :)

I also change the box if.